### DISCUSSION TITLE
Problems in the detections of some devices.

### DISCUSSION THREAD

**Discussion by PierreMimault** [glpi-project/glpi-agent#341] [Q&A]
### Describe the bug

Hello, my mouse and keyboard devices are displayed like this in glpi: "usb composite device", so I searched why some of the devices were displayed with the real name and others not. And I found this (see attached screenshot). So I would like to know if you have any solutions to propose to me to properly display my mice and keyboards. Note that this does not do on my webcam, headphones and usb adapters. :) (sorry for my english) 
<img width="960" alt="glpi 1" src="https://user-images.githubusercontent.com/121237270/217766869-fdbf2e70-c263-430e-bbad-8e1fbe5becbc.PNG">
<img width="960" alt="glpi 2" src="https://user-images.githubusercontent.com/121237270/217767024-ec28c318-a57d-4966-80d4-a26a22a8201e.PNG">

### To reproduce

.

### Expected behavior

.

### Operating system

Windows

### GLPI Agent version

1.4

### GLPI version

10.0.5

### GLPIInventory plugin

1.1.0

**Comment by g-bougard** [comment]
Hi @PierreMimault 

you're assuming "usb composite device" are your keyboard and mouse, but this is wrong. On computers, USB devices are indeed like a tree and this may be some part of the tree which are just there to aggregate some branch of the tree.

Even, in GLPI, you'll only see usb components which provides a serial number. And mouse and keyboard generally don't provide such information (and indeed I'm not aware of any mouse or keyboard providing such information). So this is just impossible to inventory and link such usb devices. Sometime you'll find devices with a serial number, so you see some usb devices but most of the time we don't have enough information from the system to nicely show what is the device.

In your screenshot, we can see you have some interesting USB devices. But we see nothing to link them to the devices you have in GLPI. To help you to understand, you can extract and share the `usbdevices` node from the inventory file. Or you can share the output of `glpi-inventory --partial usb` command run from an administrator console and from the installation folder of the agent. You can also share the output of `wmic path CIM_LogicalDevice get /format:list` which is a list of all logical devices on your system and the agent use this list to extract usb devices.
